lib/new_relic/agent/instrumentation/delayed_job_instrumentation.rb

```diff
@@ -83,11 +83,21 @@ def method_name(payload_object)
     defined?(::Delayed) && defined?(::Delayed::Worker) && !NewRelic::Agent.config[:disable_dj]
   end
 
+  depends_on do
+    # Because init_plugin runs dependency detection again.
+    !Delayed::Worker.method_defined?(:initialize_with_new_relic)
+  end
+
   executes do
     ::NewRelic::Agent.logger.info 'Installing DelayedJob instrumentation [part 1/2]'
   end
 
   executes do
+    unless defined?(::Delayed::Job) && ::Delayed::Job.method_defined?(:invoke_job)
+      NewRelic::Agent.logger.warn("Did not find a Delayed::Job class responding to invoke_job, aborting DJ instrumentation")
+      next
+    end
```

Comment:
I've not seen such warnings in other plugins. Should I just move it to `depends_on` block without calling a logger?